x-pack/plugins/log_explorer/public/components/flyout_detail/flyout_highlights.tsx

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useRef } from 'react';
+import { FlyoutContentActions } from '@kbn/discover-plugin/public';
+import { DataTableRecord } from '@kbn/discover-utils/src/types';
+import { FlyoutDoc } from './types';
+import * as constants from '../../../common/constants';
+import { HighlightField } from './sub_components/highlight_field';
+import {
+  cloudAccordionTitle,
+  flyoutCloudAvailabilityZoneLabel,
+  flyoutCloudInstanceIdLabel,
+  flyoutCloudProjectIdLabel,
+  flyoutCloudProviderLabel,
+  flyoutCloudRegionLabel,
+  flyoutDatasetLabel,
+  flyoutHostNameLabel,
+  flyoutLogPathFileLabel,
+  flyoutNamespaceLabel,
+  flyoutOrchestratorClusterNameLabel,
+  flyoutOrchestratorResourceIdLabel,
+  flyoutServiceLabel,
+  flyoutShipperLabel,
+  flyoutTraceLabel,
+  infraAccordionTitle,
+  otherAccordionTitle,
+  serviceAccordionTitle,
+} from './translations';
+import { HighlightSection } from './sub_components/highlight_section';
+import { DiscoverActionsProvider } from '../../hooks/use_discover_action';
+import { HighlightContainer } from './sub_components/highlight_container';
+import { useDimension } from '../../hooks/use_dimensions';
+import { useFlyoutColumnWidth } from '../../hooks/use_flyouot_column_width';
+
+export function FlyoutHighlights({
+  formattedDoc,
+  flattenedDoc,
+  actions,
+}: {
+  formattedDoc: FlyoutDoc;
+  flattenedDoc: DataTableRecord['flattened'];
+  actions: FlyoutContentActions;
+}) {
+  const elementRef = useRef<HTMLDivElement>(null);
+  const [ref, dimensions] = useDimension<HTMLDivElement>(elementRef);
```

Comment:
**suggestion:** we used quite a lot custom hooks from `react-use`, this case seems a good fit for using the [useMeasure](https://github.com/streamich/react-use/blob/master/docs/useMeasure.md) hook and avoid re-implementing a similar solution.

Reply:
Using this also implies removing the `elementRef` definition.